### PR TITLE
fix(core): skip zero mixer weights before 0*inf contribution products

### PR DESCRIPTION
**Pull request by Kayvan-Zahiri**
## Summary
- Skip zero behavior/world mixer weights before scaling prepared gradients.
- Prevents `0 * inf` from marking inactive contributions non-finite and reporting infinite used norms.

## Test plan
- [x] `pytest tests/test_representation_gradient_mixer.py -q`
- [x] `ruff check` on touched files

Made with Cursor. No Slop measured-run receipt (not an approved Codex or Claude Code client).

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by ss251 — CHANGES_REQUESTED** [review]
AI provider/model: OpenAI gpt-5.6-sol via Codex 0.149.0

Reviewed exact head `7652e9000497e1103c7c9b68960553829ec97d92`.

Disposition: **CLOSE**

The claimed `0 * inf` contribution cannot arise from the supported mixer path. Before `_prepare_source`, non-finite raw entries are replaced with zero; with finite input, normalization and clipping return finite values. I exercised base with float32-maximum inputs under both supported normalizations and with raw infinities:

```text
2616 finite-limit: none valid= True behavior_used_norm= 0.0 gradient= [3.0, -4.0]
2616 finite-limit: unit_l2 valid= True behavior_used_norm= 0.0 gradient= [3.0, -4.0]
2616 nonfinite raw is sanitized: valid= True behavior_used_norm= 0.0 gradient= [3.0, -4.0]
```

The regression creates the failure only by monkeypatching `_prepare_source` to return infinities, violating that internal invariant. Moreover, `_skip_zero_scale` at `alberta_framework/core/representation_gradient_mixer.py:536-539` uses `jnp.where(scale == 0, ..., scale * value)`, so JAX still evaluates the product branch; it masks the result rather than literally skipping evaluation.

Exact-head execution:

```text
$ python -m pytest tests/test_representation_gradient_mixer.py -q -o addopts= -p no:cacheprovider
........................................................................ [ 98%]
.                                                                        [100%]
73 passed in 3.27s
```

Green tests do not rescue the premise: no supported input reproduces an infinite prepared inactive source, and the added helper does not perform the claimed execution skip.

```slop-review
{"schemaVersion":"2","projectId":"asi","artifactUrl":"https://github.com/elizaOS/asi/pull/2616","headSha":"7652e9000497e1103c7c9b68960553829ec97d92","provider":"openai","model":"gpt-5.6-sol","client":"codex","runId":"run_01M11F98H8YMC16M5TA6S30Z4J","traceSha256":"24e59b6803c4280d3919cec300172d5211dae0099b32a6fbf49fcb0a29d697de","recommendation":"reject","reproduced":false,"securityRisk":"none","duplicateRisk":"none","splitRisk":"none","effortBand":"micro","complexity":"low","impact":"narrow","reviewLoad":"standard","recommendedTier":"micro","recommendedThirds":1,"workUnitId":"wu_asi_mixer_monkeypatched_invariant_2616","confidenceBasisPoints":9950,"valueRationale":"Base remains finite at float32 limits and sanitizes raw infinities; only monkeypatching a private helper creates the claimed state, and jnp.where still evaluates the product branch.","usefulArtifacts":[],"commands":["python -m pytest tests/test_representation_gradient_mixer.py -q -o addopts= -p no:cacheprovider"],"evidenceUrls":["https://github.com/elizaOS/asi/pull/2616"],"summary":"CLOSE at 7652e900: 73 tests pass, but finite-limit and nonfinite-raw supported probes already produce exact zero, while the regression violates a private invariant."}
```

Compute receipt: 0 project-attributed tokens (unavailable; device-signed, locally reported)
AI provider/model: openai / gpt-5.6-sol
Client / agent tooling: codex
Contribution skill revision: SlopDotCash/slopdotcash@ffc07b0ca4888df20faee5b490181dbcaae5cc91:skills/contribute-to-asi
Attribution status: self-reported
— [review-2616-2623]

**Comment by lalalune** [comment]
Closing after implementation and regression review. The test replaces _prepare_source with a monkeypatch that returns infinity for a finite vector. The real source preparation uses max-scaled normalization/clipping and finite source guards; this artificial replacement does not reproduce a reachable defect. The added helper does not resolve an integration gap.
